src/main/java/io/codelex/flightplanner/flightmanager/service/FlightServiceDB.java

```diff
@@ -0,0 +1,110 @@
+package io.codelex.flightplanner.flightmanager.service;
+
+import io.codelex.flightplanner.flightmanager.domain.Airport;
+import io.codelex.flightplanner.flightmanager.domain.Flight;
+import io.codelex.flightplanner.flightmanager.dto.PageResult;
+import io.codelex.flightplanner.flightmanager.dto.SearchFlightRequest;
+import io.codelex.flightplanner.flightmanager.repository.AirportRepository;
+import io.codelex.flightplanner.flightmanager.repository.FlightRepository;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@ConditionalOnProperty(prefix = "flight-app", name = "storage", havingValue = "database")
+public class FlightServiceDB implements FlightService {
+    private final FlightRepository flightRepository;
+    private final AirportRepository airportRepository;
+
+    public FlightServiceDB(AirportRepository airportRepository, FlightRepository flightRepository) {
+        this.airportRepository = airportRepository;
+        this.flightRepository = flightRepository;
+    }
+
+    @Override
+    public void clearFlights() {
+        flightRepository.deleteAll();
+    }
+
+    @Transactional
+    public Airport checkAirport(Airport airport) {
+        Optional<Airport> airportOptional = airportRepository.findById(airport.getAirport());
+        return airportOptional.orElseGet(() -> airportRepository.save(airport));
+    }
+
+    @Override
+    @Transactional
+    public void addFlight(Flight req) {
+        req.setFrom(checkAirport(req.getFrom()));
+        req.setTo(checkAirport(req.getTo()));
+        flightRepository.save(req);
+    }
+
+    @Override
+    @Transactional
+    public synchronized void deleteFlight(Long id) {
+        if (flightRepository.existsById(id)) {
+            flightRepository.deleteById(id);
+        }
+    }
+
+    @Override
+    public boolean isNotRepeated(Flight flight) {
+        return flightRepository.findAll().stream().noneMatch(f -> f.equals(flight));
```

Comment:
Ar findAll lietošanu DB gadījumā ir jāuzmanās - visbiežāk to nedrīkst lietot.
Ja DB būtu 10 000 ieraksti, tad viņi visu tiktu ielasīti atmiņā un tas var vienā brīdī nokārt sistēmu. Bet ir iespējams uzreiz repo meklēt pēc konkrētiem kritērijiem, vai rakstīt custom meklēšanu, kas notiek DB pusē.
Izlasi šito un arī zemāk tajā pašā rakstā ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries
Kā ari ir šāds vairants:
https://www.baeldung.com/spring-data-query-by-example#examples